src/afrolabs/components/kafka.clj

```diff
@@ -1606,11 +1609,19 @@
                                             (str t) (str k) (str v)))
                               old)))
                         (fn [old-meta]
-                          (if-not (and o p)
-                            old-meta
-                            (update-in old-meta [:ktable/topic-partition-offsets t p]
+                          (cond-> old-meta
+                            (and o p)
+                            (update-in [:ktable/topic-partition-offsets t p]
                                        (fnil #(max % o)
-                                             -1)))))
+                                             -1))
+
+                            hdrs
+                            (assoc-in [:ktable/record-headers t k]
+                                      hdrs)
+
+                            (not hdrs)
+                            (update-in [:ktable/record-headers t]
+                                       #(dissoc % k)))))
```

Comment:
We now also store the headers on individual records centrally (and not only on the record values themselves). This helps, because some records have nil-values while still carying headers. If we have aggregations that need to resume based on header info, then we need to retain the header values for nil-valued records, so we keep it in this central meta-data on the ktable value itself.